src/transformers/models/oneformer/configuration_oneformer.py

```diff
@@ -0,0 +1,205 @@
+# coding=utf-8
+# Copyright 2022 SHI-Labs Copyright 2022 The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""OneFormer model configuration"""
+import copy
+from typing import Dict, Optional
+
+from transformers import SwinConfig
+
+from ...configuration_utils import PretrainedConfig
+from ...utils import logging
+from ..auto import CONFIG_MAPPING
+
+
+logger = logging.get_logger(__name__)
+
+ONEFORMER_PRETRAINED_CONFIG_ARCHIVE_MAP = {
+    "shi-labs/oneformer_ade20k_swin_tiny": (
+        "https://huggingface.co/shi-labs/oneformer_ade20k_swin_tiny/blob/main/config.json"
+    ),
+    # See all OneFormer models at https://huggingface.co/models?filter=oneformer
+}
+
+
+class OneFormerConfig(PretrainedConfig):
+    r"""
+    This is the configuration class to store the configuration of a [`OneFormerModel`]. It is used to instantiate a
+    OneFormer model according to the specified arguments, defining the model architecture. Instantiating a
+    configuration with the defaults will yield a similar configuration to that of the OneFormer
+    [shi-labs/oneformer_ade20k_swin_tiny](https://huggingface.co/shi-labs/oneformer_ade20k_swin_tiny) architecture
+    trained on [ADE20k-150](https://huggingface.co/datasets/scene_parse_150).
+
+    Configuration objects inherit from [`PretrainedConfig`] and can be used to control the model outputs. Read the
+    documentation from [`PretrainedConfig`] for more information.
+
+    Currently, OneFormer supports the [Swin Transformer](swin) and [Dilated Neighborhood Attention Transformer](dinat)
+    as backbones.
+
+    Args:
+        output_attentions (`bool`, *optional*, defaults to `True`):
+            Whether to output attention weights.
+        output_hidden_states (`bool`, *optional*, defaults to `True`):
+            Whether to output intermediate predictions.
+        return_dict (`bool`, *optional*, defaults to `True`):
+            Whether to return output as tuples or dataclass objects.
+        general_config (`dict`, *optional*)
+            Dictionary containing general configuration like backbone_type, loss weights, number of classes, etc.
```

Comment:
Rather than having a single general_config argument, we prefer to have separate arguments for each of the hyperparameters that define a model architecture. See `configuration_detr.py` (or any other model) as an example.